examples/http/aws_rds_metrics.js

```diff
@@ -0,0 +1,511 @@
+        date = amzdate.replace(/T\d+Z/, ""),
+        host = service + "." + region + ".amazonaws.com:443",
+        device = D.createExternalDevice(service + "." + region + ".amazonaws.com"),
+        canonicalUri = "/",
+        canonicalHeaders = "content-encoding:amz-1.0\n" + "host:" + host + "\n" + "x-amz-date:" + amzdate + "\n",
+        signedHeaders = "content-encoding;host;x-amz-date",
+        canonicalRequest = method + "\n" + canonicalUri + "\n" + "\n" + canonicalHeaders + "\n" + signedHeaders + "\n" + sha256(body),
+        credentialScope = date + "/" + region + "/" + service + "/" + "aws4_request",
+        requestString = "AWS4-HMAC-SHA256" + "\n" + amzdate + "\n" + credentialScope + "\n" + sha256(canonicalRequest),
+        key = sign("AWS4" + secretKey, date);
+    key = sign(key, region);
+    key = sign(key, service);
+    key = sign(key, "aws4_request");
+    var auth = "AWS4-HMAC-SHA256 Credential=" + accessKey + "/" + credentialScope + ", " + "SignedHeaders=" + signedHeaders + ", " + "Signature=" + hmac("sha256", key, requestString);
+    device.http.post({
+        url: canonicalUri,
+        protocol: "https",
+        headers: {
+            "x-amz-date": amzdate,
+            "X-Amz-Target": "GraniteServiceVersion20100801.GetMetricData",
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Content-Encoding": "amz-1.0",
+            "Authorization": auth
+        },
+        body: body
+    },
+        function (err, response, body) {
+            if (err) {
+                D.failure(D.errorType.GENERIC_ERROR);
+            }
+            if (response.statusCode == 404) {
+                D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+            }
+            if (response.statusCode == 401) {
+                D.failure(D.errorType.AUTHENTICATION_ERROR);
+            }
+            if (response.statusCode != 200) {
+                D.failure(D.errorType.GENERIC_ERROR);
+            }
+            d.resolve(JSON.parse(body));
+        });
+    return d.promise;
+}
+
+/**
+ * @returns promise for http response containing instance metrics 
+ */
+function getMetricsData() {
+    var payload = {},
+        endTime = Math.floor((new Date().getTime()) / 1000),
+        startTime = endTime - requestPeriod;
+    payload["StartTime"] = startTime;
+    payload["EndTime"] = endTime;
+    payload["ScanBy"] = "TimestampDescending";
+    payload["MetricDataQueries"] = createMetricsPayload(requestPeriod, dbInstanceId);
+    return httpPost(payload)
+        .then(function (data) {
+            metrics = data.MetricDataResults;
+        });
+}
+
+/**
+ * @param {string} label  The label to search for in the metric array.
+ * @returns A function that takes no arguments and returns the first value associated with the given label, or null if no such label exists.
+ */
+function extractValue(label) {
```

Comment:
add rounding (to 3rd decimal or sth) to avoid such long values for numerical only:

249176.49038781217